inception/inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasDocumentIndex.java

```
@@ -396,6 +402,221 @@ public long numberOfQueryResults(SearchQueryRequest aRequest)
+        results.remove("n");
+        allStats.put("per Sentence: Token Count", results);
+        nonNullStats.put("per Sentence: Token Count", results);
+
+        results = getLayerStatistics(aStatisticRequest, "<s=\"\"/>", fullDocSet, false);
+        results.remove("stats");
+        results.remove("sourceNumber");
+        results.put("Number of Documents", results.get("n"));
+        results.remove("n");
+        allStats.put("Sentence Count", results);
+        nonNullStats.put("Sentence Count", results);
+
+        return new StatisticsResult(aStatisticRequest, allStats, nonNullStats);
+    }
+
+    @Override
+    public ArrayList<Integer> getUniqueDocuments(StatisticRequest aStatisticRequest)
+        throws IOException
+    {
+        IndexSearcher searcher = null;
+        Map<Long, Long> annotatableDocuments = listAnnotatableDocuments(
+                aStatisticRequest.getProject(), aStatisticRequest.getUser());
+
+        ArrayList<Integer> fullDocSet = new ArrayList<Integer>();
+
+        try {
+            searcher = getSearcherManager().acquire();
+            IndexReader reader = searcher.getIndexReader();
+
+            for (int i = 0; i < reader.maxDoc(); i++) {
+                String sourceID = reader.document(i).get(FIELD_SOURCE_DOCUMENT_ID);
+                String annotationID = reader.document(i).get(FIELD_ANNOTATION_DOCUMENT_ID);
+                if (Long.valueOf(annotationID) != -1L) {
+                    if (reader.document(i).get(FIELD_USER)
+                            .equals(aStatisticRequest.getUser().getUsername())) {
+                        fullDocSet.add(i);
+                    }
+                }
+                // source document without annotation layer? then user is not relevant
+                else if (!annotatableDocuments.containsKey(Long.valueOf(sourceID))) {
+                    fullDocSet.add(i);
+                }
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```

This needs a bit more care, swallowing exceptions is not good